template/enum_types_header.njk

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2017 The Bacardi Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GEN_ENUM_TYPES_H_
+#define GEN_ENUM_TYPES_H_
+
+#include <set>
+#include <string>
+
+class EnumTypes {
```

Comment:
IMHO, we don't need to generate this type.
We can just pass the string set into isValidEnum() directly.

Reply:
do you means to pass the string set into isVaildEnum() directly in interface_cpp.njk?

Reply:
Yeah, exactly!

Reply:
i changed to pass to the string set as parameter.